iron-focus-wrap-behavior.html

```diff
@@ -0,0 +1,236 @@
+      }
+
+      if (shouldWrap) {
+        // When the element contains the last focusable element of the document
+        // and it's already focused, pressing TAB would move the focus outside
+        // the document (e.g. to the browser search bar). Similarly, when the
+        // element contains the first focusable element of the document and it's
+        // already focused, pressing Shift+TAB would move the focus outside the
+        // document (e.g. to the browser search bar).
+        // In both cases, we would not receive a focus event, but only a blur.
+        // In order to achieve focus wrapping, we prevent this TAB event and
+        // force the focus. This will also prevent the focus to temporarily move
+        // outside the element, which might cause scrolling.
+        event.preventDefault();
+        nodeToSet.focus();
+      }
+    },
+
+    /**
+     * The deepest active element.
+     * @type {!Element} activeElement
+     */
+    __getDeepActiveElement: function() {
+      // document.activeElement can be null
+      // https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement
+      // In case of null, default it to document.body.
+      var active = document.activeElement || document.body;
+      while (active.root && Polymer.dom(active.root).activeElement) {
+        active = Polymer.dom(active.root).activeElement;
+      }
+      return active;
+    },
+
+    /**
+     * Array of nodes that can receive focus (host included), ordered by `tabindex`.
+     * @param {boolean} searchInShadowDOM If true, the search is performed in the
+     * shadow dom.
+     * @returns {Array<Node>}
+     * @private
+     */
+    __getFocusableNodes: function(searchInShadowDOM) {
+      // Elements that can be focused even if they have [disabled] attribute.
+      var FOCUSABLE_WITH_DISABLED = [
+        'a[href]',
+        'area[href]',
+        'iframe',
+        '[tabindex]',
+        '[contentEditable=true]'
+      ];
+
+      // Elements that cannot be focused if they have [disabled] attribute.
+      var FOCUSABLE_WITHOUT_DISABLED = [
+        'input',
+        'select',
+        'textarea',
+        'button'
+      ];
+
+      // Discard elements with tabindex=-1 (makes them not focusable).
+      var selector = FOCUSABLE_WITH_DISABLED.join(':not([tabindex="-1"]),') +
+        ':not([tabindex="-1"]),' +
+        FOCUSABLE_WITHOUT_DISABLED.join(':not([disabled]):not([tabindex="-1"]),') +
+        ':not([disabled]):not([tabindex="-1"])';
+
+      var node = searchInShadowDOM ? this.root : this;
+      var focusables = Polymer.dom(node).querySelectorAll(selector);
+      if (this.tabIndex >= 0) {
+        // Insert at the beginning because we might have all elements with tabIndex = 0,
+        // and the overlay should be the first of the list.
+        focusables.splice(0, 0, this);
+      }
+      // Sort by tabindex.
+      return focusables.sort(function (a, b) {
+        if (a.tabIndex === b.tabIndex) {
```

Comment:
Actually, that would not help, since the keyboard event listener is on the element itself, and we don't know where the focus will go after the TAB is done...I'd defer the support of that use case to be handled outside this behavior, WDYT?